crates/bevy_reflect/src/array.rs

```diff
@@ -303,8 +300,18 @@ impl Typed for DynamicArray {
 
 /// An iterator over an [`Array`].
 pub struct ArrayIter<'a> {
-    pub(crate) array: &'a dyn Array,
-    pub(crate) index: usize,
+    array: &'a dyn Array,
+    index: usize,
+}
+
+impl<'a> ArrayIter<'a> {
+    /// Creates a new [`ArrayIter`].
+    ///
+    /// It is an [`Iterator`] for [`Array`] whose elements start from `index`.
```

Comment:
Nit: Not sure this part is needed as it's essentially just restating the docs on `ArrayIter` itself. Additionally, `index` has been removed as a parameter, making that part unneeded as well.
```suggestion
```